For OpenJS Review: `vis.gl` Charter and Governance Proposal for OpenJS (#15)
visgl/tsc: CHARTER.md

# vis.gl Charter

> Vis.gl aims to provide an open-source JavaScript framework for high-performance visualizations of large datasets on the web. It prioritizes strong support for geospatial data visualization, while also being flexible to cater to diverse non-geospatial visualization needs. It is designed to make advanced visualization techniques accessible to a wide range of developers by providing a user-friendly API and well-documented framework.

Reviewer: ```suggestion
Vis.gl aims to provide an open-source JavaScript framework for high-performance visualizations of large datasets on the web. It prioritizes strong support for geospatial data visualization, while also being flexible enough to cater to diverse non-geospatial visualization needs. It is designed to make advanced visualization techniques accessible to a wide range of developers by providing a user-friendly API and well-documented framework.
```

## Section 0: Guiding Principles

> We believe that powerful visualization should be accessible to every developer.

We believe visualization tools should be declarative, composable, and interoperable—accessible to every developer and designed to work seamlessly within the broader ecosystem.

> Power shouldn’t come from complexity: layers, not low-level rendering; clear abstractions, not siloed systems. Our libraries should fit naturally alongside the tools developers already use, while staying modular and focused so each part can stand alone or be combined into a coherent whole.

Author: @felixpalmer @ibgreen @Pessimistress,

We previously had a copy/pasta under guiding principles and the CPC has requested (see https://github.com/visgl/tsc/pull/15#discussion_r2470738743) we put something that we believe instead. Any objections to this wording?

Section Intentionally Left Blank  

### 1.2: Out-of-Scope

Section Intentionally Left Blank

## Section 2: Relationship with OpenJS Foundation CPC.

The OpenJS Cross Project Council (CPC) delegates technical leadership of this project to the governing body defined in Section 3 of this charter.

This project is entitled to representation in the CPC through voting members as described in [Section 4](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#voting-members) of the CPC Charter.

### 2.1 Other Formal Project Relationships

Section Intentionally Left Blank

## Section 3: `vis.gl`'s TSC Governing Body

`vis.gl` is governed by its [Technical Steering Committee](https://github.com/visgl/tsc/blob/master/README.md#technical-steering-committee).

## Section 4: Roles & Responsibilities 

The responsibilities of `vis.gl`'s TSC are described in [Section 6](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-6-non-responsibilities-of-the-cpc) of the CPC Charter.

### Section 4.1 Project Operations & Management

Section Intentionally Left Blank

### Section 4.2: Decision-making, Voting, and/or Elections

> Decision making and voting follow the practices adopted by the CPC and described in [Section 9](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-9-decision-making) and [Section 10](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-10-voting) of the CPC Charter respectively.

Author: > Pulled up from an outdated comment

@felixpalmer @ibgreen @Pessimistress,

After discussion with @tobie, it's more accurate to say use lazy consensus for decisions.

Lazy consensus is designed to be a fast-ish form of governance. Faster than full consensus (especially in large TSCs), and way faster than voting on every little thing.

Lazy consensus implies there's a reasonable amount of time between proposal and action so that objections can be made.

If we care about _how much time_, we can put this timing in our governance doc (e.g. 2 weeks for important decisions and a couple days for minor changes).

If we care about how many people need to agree, we put that in our governance doc (e.g. 4 members need to agree to adding a new TSC member, or sub-project).

The benefit to all here is that the TSC can edit their governance without approval from the CPC.

Last resort voting, as described by CPC in section 10, is fine.

Any objections to this?

# vis.gl Charter

Vis.gl aims to provide an open-source JavaScript framework for high-performance visualizations of large datasets on the web. It prioritizes strong support for geospatial data visualization, while also being flexible to cater to diverse non-geospatial visualization needs. It is designed to make advanced visualization techniques accessible to a wide range of developers by providing a user-friendly API and well-documented framework.

## Section 0: Guiding Principles

> The `vis.gl` project is part of the [OpenJS Foundation](https://openjsf.org), which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.

Author: Rewritten into a belief statement

`vis.gl` is to provide a comprehensive, open-source solution for GPU-powered visualizations of large-scale datasets, in particular geospatial data visualizations. The `vis.gl` project consists of many different sub-projects, such as a layer-based visualization framework, GPU toolkit, basemap react components, and more. The project is designed to be modular, so that it can be used in a variety of different environments and for different purposes.

### 1.1: In-scope

 - Overseeing the **technical development** of the core vis.gl libraries and all affiliated sub-projects.
 - **Release planning** and coordination
 - Managing project repositories and **governance**, including issue tracking, pull request processes, and repository organization.
 - Defining and enforcing **coding standards, testing practices, and development processes** to maintain code quality.
 - Maintaining the list of **Maintainers and core contributors**, and setting permissions in project repositories.
 - Reviewing and proposing **new project incubations or sub-projects**, as well as deciding on deprecations or significant changes to the project’s scope.
 - Formation of technical **working groups**.

>  - **Representation** in technical collaborations with external communities.

Reviewer 2: ```suggestion
 Section Intentionally Left Blank
```

This is about the project's scope, not the TSC's responsibilities, so "problem spaces, use cases, repositories or other projects which are included with the work but may not be readily apparent."

This `vis.gl` Charter reflects a carefully constructed balanced role for the TSC and the CPC in the governance of the OpenJS Foundation. The charter amendment process is for the TSC to propose changes using simple majority of the full TSC, the proposed changes being subject to review and approval by the CPC. The CPC may additionally make amendments to the project charter at any time, though the CPC will not interfere with day-to-day discussions, votes, or meetings of the TSC.

### 2.1 Other Formal Project Relationships

Section Intentionally Left Blank

## Section 3: `vis.gl`'s TSC Governing Body

`vis.gl` is governed by its [Technical Steering Committee](https://github.com/visgl/tsc/blob/master/README.md#technical-steering-committee).

## Section 4: Roles & Responsibilities 

> The roles and responsibilities of `vis.gl`'s TSC are described in [GOVERNANCE.md](./GOVERNANCE.md).

Reviewer 2: ```suggestion
The responsibilities of `vis.gl`'s TSC are described in [Section 6](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-6-non-responsibilities-of-the-cpc) of the CPC Charter.

```

### Section 4.1 Project Operations & Management

Section Intentionally Left Blank

### Section 4.2: Decision-making, Voting, and/or Elections

> The vis.gl TSC operates on a [consensus-seeking basis](https://en.wikipedia.org/wiki/Consensus-seeking_decision-making). Most decisions are expected to be resolved through open discussion and agreement. However, when consensus cannot be reached, any TSC member may call for a vote.

Reviewer 2: Suggest adopting the default language that's in the template charter ([Section 4.2](https://github.com/openjs-foundation/cross-project-council/blob/tobie-updated-template-charter/PROJECT_CHARTER_TEMPLATE.md#section-42-decision-making-voting-andor-elections)).

 - **Eligibility:** All steering committee members are voting members and may participate in any vote.

 - **Quorum:** A quorum is established when at least 50% of steering committee members members are present. The steering committee members may meet without quorum, but may not make binding decisions.

>  - **Votes During Meetings:** When quorum is met, decisions require a simple majority of the steering committee members present.

Author: After discussion it's more accurate to say use lazy consensus for decisions. The key is that lazy requires time between proposal and action to make decisions - and we put this timing in our goverance doc (e.g. 2 weeks for important decisions and a couple days for minor changes). Lazy also allows a goverance can specify how many people need to agree for speicic deciisons to be made (e.g. adding a new TSC member, or sub-project). Last resort voting, as described by CPC, is fine.

 - **Electronic Votes:** Decisions made asynchronously (e.g., over email or GitHub) require a simple majority of all steering committee members.

>  - **Escalation:** If a vote results in a tie or fails to resolve a dispute, any TSC member may refer the matter to the OpenJS Foundation Board for mediation or guidance.

Reviewer 2: We'd want the TSC chair (not any TSC member) to raise this with the CPC (not the Board).
